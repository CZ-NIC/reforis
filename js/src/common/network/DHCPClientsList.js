/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

DHCPClientsList.propTypes = {
    clients: PropTypes.arrayOf(PropTypes.object),
};

export default function DHCPClientsList({ clients }) {
    return (
            <h2>{_("DHCP Client List")}</h2>
            <p>
                {_(
                    "This list contains all devices that are connected to the network through wired or wireless connections."
                )}
            </p>
            {clients.length === 0 ? (
                <p className="text-muted text-center">
                    {_("No DHCP clients found.")}
                </p>
            ) : (
                <div className="table-responsive">
                    <table className="table table-hover">
                        <thead>
                            <tr className="text-center">
                                <th>{_("Expires")}</th>
                                <th>{_("IP Address")}</th>
                                <th>{_("MAC Address")}</th>
                                <th>{_("Hostname")}</th>
                                <th>{_("Active")}</th>
                            </tr>
                        </thead>
                        <tbody>
                            {clients.map((client) => (
                                <DHCPClientsListItem
                                    key={client.ip}
                                    {...client}
                                />
                            ))}
                        </tbody>
                    </table>
                </div>
            )}
        </>
    );
}

DHCPClientsListItem.propTypes = {
    ip: PropTypes.string.isRequired,
    expires: PropTypes.number.isRequired,
    mac: PropTypes.string.isRequired,
    hostname: PropTypes.string.isRequired,
    active: PropTypes.bool.isRequired,
};

function DHCPClientsListItem({ ip, expires, mac, hostname, active }) {
    return (
        <tr className="text-center">
            <td>{moment.unix(expires).format("YYYY-MM-DD HH:mm")}</td>
            <td>{ip}</td>
            <td>{mac}</td>
            <td>{hostname}</td>
            <td>
                <i className={`fas ${active ? "fa-check" : "fa-times"}`} />
            </td>
        </tr>
    );
}
