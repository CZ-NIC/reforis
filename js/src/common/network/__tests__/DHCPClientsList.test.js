/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import {render} from '@testing-library/react';


import DHCPClientsList from '../DHCPClientsList';
import {clients} from './__fixtures__/DHCPClientsList';

describe('<DHCPClientsList/>', () => {
    it('Test with snapshot.', () => {
        const {container} = render(<DHCPClientsList clients={clients}/>);
        expect(container).toMatchSnapshot();
    })
});
