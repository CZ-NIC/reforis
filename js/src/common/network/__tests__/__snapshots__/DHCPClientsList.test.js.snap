// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DHCPClientsList/> Test with snapshot. 1`] = `
<div>
  <h2>
    DHCP Clients
  </h2>
  <div
    class="table-responsive"
  >
    <table
      class="table table-hover"
    >
      <thead>
        <tr
          class="text-center"
        >
          <th>
            Expires
          </th>
          <th>
            IP Address
          </th>
          <th>
            MAC Address
          </th>
          <th>
            Hostname
          </th>
          <th>
            Active
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="text-center"
        >
          <td>
            2018-10-12 13:16
          </td>
          <td>
            192.168.1.1
          </td>
          <td>
            11:22:33:44:55:66
          </td>
          <td>
            first
          </td>
          <td>
            <i
              class="fas fa-check"
            />
          </td>
        </tr>
        <tr
          class="text-center"
        >
          <td>
            2018-10-12 13:16
          </td>
          <td>
            192.168.2.1
          </td>
          <td>
            99:88:77:66:55:44
          </td>
          <td>
            *
          </td>
          <td>
            <i
              class="fas fa-times"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
